cupy_backends/cuda/libs/cusparse.pyx

```diff
@@ -1572,45 +1572,46 @@ cpdef intptr_t create() except? 0:
     return <intptr_t>handle
 
 
-cpdef size_t createMatDescr():
+cpdef size_t createMatDescr() except? -1:
     cdef MatDescr desc
     status = cusparseCreateMatDescr(& desc)
     check_status(status)
     return <size_t>desc
 
 
-cpdef destroy(intptr_t handle):
+cpdef void destroy(intptr_t handle) except *:
```

Comment:
For functions returning void, an exception state must always be checked.

Reply:
It seems the only significant difference is `__Pyx_INCREF(Py_None)` call being removed, so there seems no much difference. But I still think it's better to type-anontate everything, rather than maintaining a complex coding rule like "`except` can be omitted if return type is not annotated or is a Python object".

Without type annotation:

```cpp
static PyObject *__pyx_pw_13cupy_backends_4cuda_4libs_8cusparse_11destroy(PyObject *__pyx_self, PyObject *__pyx_arg_handle); /*proto*/
static PyObject *__pyx_f_13cupy_backends_4cuda_4libs_8cusparse_destroy(intptr_t __pyx_v_handle, CYTHON_UNUSED int __pyx_skip_dispatch) {
  cusparseStatus_t __pyx_v_status;
  PyObject *__pyx_r = NULL;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("destroy", 0);
/* … */
  /* function exit code */
  __pyx_r = Py_None; __Pyx_INCREF(Py_None);
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("cupy_backends.cuda.libs.cusparse.destroy", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_r = 0;
  __pyx_L0:;
  __Pyx_XGIVEREF(__pyx_r);
  __Pyx_RefNannyFinishContext();
  return __pyx_r;
}
```

With annotation:

```cpp
static PyObject *__pyx_pw_13cupy_backends_4cuda_4libs_8cusparse_11destroy(PyObject *__pyx_self, PyObject *__pyx_arg_handle); /*proto*/
static void __pyx_f_13cupy_backends_4cuda_4libs_8cusparse_destroy(intptr_t __pyx_v_handle, CYTHON_UNUSED int __pyx_skip_dispatch) {
  cusparseStatus_t __pyx_v_status;
  __Pyx_RefNannyDeclarations
  __Pyx_RefNannySetupContext("destroy", 0);
/* … */
  /* function exit code */
  goto __pyx_L0;
  __pyx_L1_error:;
  __Pyx_XDECREF(__pyx_t_1);
  __Pyx_AddTraceback("cupy_backends.cuda.libs.cusparse.destroy", __pyx_clineno, __pyx_lineno, __pyx_filename);
  __pyx_L0:;
  __Pyx_RefNannyFinishContext();
}
```